ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts.py

```diff
@@ -385,11 +395,47 @@ def uplink_switches(self):
 
     @cached_property
     def uplink_interfaces(self):
-        return get(self._switch_data_combined, "uplink_interfaces")
+        return self._range_expand(default(
+            get(self._switch_data_combined, "uplink_interfaces"),
+            get(self.platform_settings, "default_uplink_interfaces"),
+            [])
+        )
 
     @cached_property
     def uplink_switch_interfaces(self):
-        return get(self._switch_data_combined, "uplink_switch_interfaces")
+        uplink_switch_interfaces = get(self._switch_data_combined, "uplink_switch_interfaces")
+        if uplink_switch_interfaces is not None:
+            return uplink_switch_interfaces
+
+        if self.uplink_switches is None:
+            return []
+
+        uplink_switch_interfaces = []
+        for uplink_switch_index, uplink_switch in enumerate(self.uplink_switches):
+            uplink_switch_facts: EosDesignsFacts = get(self._hostvars,
+                                                       f"avd_switch_facts..{uplink_switch}..switch",
+                                                       required=True,
+                                                       org_key=f"avd_switch_facts.({uplink_switch}).switch",
+                                                       separator="..")
+
+            # Create a list of indexes of uplink_switches where it is this uplink_switch (parallel uplinks)
+            indices_of_uplinks_to_uplink_switch = [index for index, switch in enumerate(self.uplink_switches) if switch == uplink_switch]
+            # Out of the parallel uplinks, where is this uplink?. Normally this would be 0 for single uplinks.
+            index_of_parallel_uplinks = indices_of_uplinks_to_uplink_switch.index(uplink_switch_index)
```

Comment:
Looks good but this is doing too many for loops.
```suggestion
        uplink_switch_counter = {}
        for uplink_switch in self.uplink_switches:
            uplink_switch_facts: EosDesignsFacts = get(self._hostvars,
                                                       f"avd_switch_facts..{uplink_switch}..switch",
                                                       required=True,
                                                       org_key=f"avd_switch_facts.({uplink_switch}).switch",
                                                       separator="..")

            # Count the number of instances the current switch was processed
            uplink_switch_counter[uplink_switch] = uplink_switch_counter.get(uplink_switch, 0) + 1
            index_of_parallel_uplinks = uplink_switch_counter[uplink_switch] - 1
```

I believe the dictionary get method should work or we can use:

```
uplink_switch_counter[uplink_switch] = get(uplink_switch_counter, uplink_switch, default=0) + 1
```

Your, opinion @ClausHolbechArista

Reply:
In fact, we can use default value as -1 for uplink_switch_counter[uplink_switch] instead of 0, but as I said it's a counter, I did it this way. What ever looks good, use it.

Reply:
This looks good to me - thanks.